imxrt-hal/src/tempmon.rs

```diff
@@ -0,0 +1,395 @@
+
+        let n1_room_count = (calibration >> 20) as i32;
+        let t1_room_temp = 25_000_i32;
+        let n2_hot_count = ((calibration >> 8) & 0xFFF) as i32;
+        let t2_hot_temp = (calibration & 0xFF) as i32 * 1_000;
+
+        // Tmeas = HOT_TEMP - (Nmeas - HOT_COUNT) * ((HOT_TEMP - 25.0) / (ROOM_COUNT – HOT_COUNT))
+        let scaler = (t2_hot_temp - t1_room_temp) / (n1_room_count - n2_hot_count);
+
+        // Tmeas = HOT_TEMP - (Nmeas - HOT_COUNT) * scaler
+
+        TempMon {
+            base: self.0,
+            scaler,
+            hot_count: n2_hot_count,
+            hot_temp: t2_hot_temp,
+            measurement_active: false,
+        }
+    }
+
+    /// Initialize the temperature monitor.
+    ///
+    /// The `measure_freq` determines how many RTC clocks to wait before automatically repeating a temperature
+    /// measurement. The pause time before remeasuring is the field value multiplied by the RTC period.
+    ///
+    /// Find more details `TempMon.set_measure_frequency`
+    pub fn init_with_measure_freq(self, measure_freq: u16) -> TempMon {
+        let mut t = self.init();
+        t.set_measure_frequency(measure_freq);
+        t
+    }
+}
+
+/// A Temperature Monitor (TEMPMON)
+///
+/// # Example 1
+///
+/// ```no_run
+/// use imxrt1060_hal;
+///
+/// let mut peripherals = imxrt1060_hal::Peripherals::take().unwrap();
+/// let (_, ipg_hz) = peripherals.ccm.pll1.set_arm_clock(
+///     imxrt1060_hal::ccm::PLL1::ARM_HZ,
+///     &mut peripherals.ccm.handle,
+///     &mut peripherals.dcdc,
+/// );
+/// let mut cfg = peripherals.ccm.perclk.configure(
+///     &mut peripherals.ccm.handle,
+///     imxrt1060_hal::ccm::perclk::PODF::DIVIDE_3,
+///     imxrt1060_hal::ccm::perclk::CLKSEL::IPG(ipg_hz),
+/// );
+///
+/// // init temperature monitor
+/// // consider using init_with_measure_freq
+/// let mut temp_mon = peripherals.tempmon.init();
+/// loop {
+///     if let Ok(_temperature) = nb::block!(temp_mon.measure_temp()) {
+///         // _temperature in mC (1°C = 1000°mC)
+///     }
+/// }
+/// ```
+pub struct TempMon {
+    base: ral::tempmon::Instance,
+    /// scaler * 1000
+    scaler: i32,
+    /// hot_count * 1000
+    hot_count: i32,
+    /// hot_temp * 1000
+    hot_temp: i32,
+
+    /// measurement active flag to get measure_temp unblocking
+    measurement_active: bool,
+}
+
+impl TempMon {
+    /// converts the temp_cnt into a human readable temperature [°mC] (1/1000 °C)
+    fn convert(&self, temp_cnt: i32) -> i32 {
+        let n_meas = temp_cnt - self.hot_count;
+        self.hot_temp - n_meas * self.scaler
+    }
+
+    /// decode the temp_value into measurable bytes
+    ///
+    /// param **temp_value_mc**: in °mC (1/1000°C)
+    ///
+    fn decode(&self, temp_value_mc: i32) -> u32 {
+        let v = (temp_value_mc - self.hot_temp) / self.scaler;
+        (self.hot_count - v) as u32
+    }
+
+    /// triggers a new measurement and waits until it's finished
```

Comment:
Are we still waiting with the adoption of `WouldBlock`?
```suggestion
    /// triggers a new measurement
```